### PR TITLE
Add Importer support

### PR DESCRIPTION
**Pull request by darkdh**
requires https://github.com/brave/libchromiumcontent/pull/3

Auditors: @bridiver, @bbondy, @bsclifton

### CODE REVIEW

**Comment by bsclifton** [comment]
@darkdh looking through these changes to understand more...

Looks like a LOT of this code is coming from Chromium, is that right? And after that, tweaks are made (functions added, edited, etc).

Since there are so many files added, would you be able to give a medium level overview of what this PR adds/changes?  Also, I'm curious what manual tests you've done and such

Reading through quickly, I can see this adds support for importing from Chrome/FF/IE and more... including dialogs associated with these.  Does this cover Chrome or FF sync import?

**Comment by darkdh** [comment]
We leverage the architecture of `ExternalProcessImporterHost` and

**Comment by darkdh** [comment]
We leverage the architecture of `ExternalProcessImporterHost` and `ExternalProcessImporterClient` with our own `ProfileWriter` since we don't use chrome's profile.
And we also add import from chrome and import cookies from firefox which both chromium doesn't support now.

All the new added files are under `atom` and files of `chromium_src` with modification will have comment in the beginning of file says `This is not a straight copy from chromium src... and what we modified`.

`atom/browser/api/atom_api_importer.*` is a good starting point which included interaction between `ExternalProcessImporterHost`, `ProfileWriter`, `ImporterList` and front end.

The manual tests I did are import history, bookmarks, cookies on chrome/firefox of different platforms, on Edge/IE of windows, on Safari of Mac.

This PR doesn't cover Chrome/Firefox sync import but it does cover different profiles of chrome.

File: filenames.gypi [hunk 1]
```diff
@@ -18,6 +18,7 @@
       'lib/browser/api/dialog.js',
       'lib/browser/api/exports/electron.js',
       'lib/browser/api/global-shortcut.js',
+      'lib/browser/api/importer.js',
```
Comment (bridiver) [inline filenames.gypi]: can you move these to an importer.gypi file? See extensions.gypi

**Comment by bridiver** [comment]
these never seem to want to close when I merge locally
++

**Pull request closed without merging**